client/status.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"context"
+
+	pb "github.com/go-phorce/trusty/api/v1/serverpb"
+	"google.golang.org/grpc"
+)
+
+type statusClient struct {
+	remote   pb.StatusClient
+	callOpts []grpc.CallOption
+}
+
+// NewStatus returns instance of Status client
+func NewStatus(conn *grpc.ClientConn, callOpts []grpc.CallOption) Status {
+	return &statusClient{
+		remote:   RetryStatusClient(conn),
+		callOpts: callOpts,
+	}
+}
+
+// NewStatusProxy returns instance of Status client
+func NewStatusProxy(remote pb.StatusClient) Status {
```

Comment:
Is not `NewStatusFromProxy` a better name? Because this creates Status interface from an already existing proxy client. That is why earlier you call it the following way. The second one makes more sense to me.

```
c.Status = client.NewStatusProxy(proxy.StatusServerToClient(s.StatusServer))
```

this should be

```
c.Status = client.NewStatusFromProxy(proxy.StatusServerToClient(s.StatusServer))
```

Reply:
agree